python/erdos/message.py

```diff
@@ -66,10 +68,10 @@ class WatermarkMessage(Message):
         timestamp: The timestamp for which this is a watermark.
     """
 
-    def __init__(self, timestamp: Timestamp):
+    def __init__(self, timestamp: Timestamp) -> None:
         super(WatermarkMessage, self).__init__(timestamp, None)
 
-    def __str__(self):
+    def __str__(self) -> T:
```

Comment:
```suggestion
    def __str__(self):
```
The `__str__` method returns a `str` and not `T`. However, I don't think we should annotate dunder methods, just because their types are usually well known.